docs/ref/templates/builtins.txt

```diff
@@ -953,6 +953,71 @@ output (as a string) inside a variable. This is useful if you want to use
     {% now "Y" as current_year %}
     {% blocktranslate %}Copyright {{ current_year }}{% endblocktranslate %}
 
+.. templatetag:: query_string
+
+``query_string``
+----------------
+
+.. versionadded:: 5.1
+
+Outputs the query string from the given :class:`~django.http.QueryDict`
+instance, or ``request.GET`` if not given and the
+``django.template.context_processors.request`` context processor is enabled.
+If the ``QueryDict`` is empty, then the output will be an empty string.
+Otherwise, the query string will be displayed with a leading "?".
+
+The following example outputs the current query string verbatim. So if the
+query string is ``?color=green&=size=M``, the output would be
+``?color=green&size=M``:
+
+.. code-block:: html+django
+
+    {% query_string %}
+
+You can also pass in a custom ``QueryDict`` that will be used instead of
+``request.GET``:
+
+.. code-block:: html+django
+
+    {% query_string my_query_dict %}
+
+Each keyword argument given will be added to the query string, replacing any
+existing value for that key. With the query string ``?color=blue``, the
+following would result in ``?color=red&size=S``:
+
+.. code-block:: html+django
+
+    {% query_string  color="red" size="S" %}
+
+It is also possible to remove values by passing ``None`` as a value. With the
+query string ``?color=blue&size=M``, the following would result in ``?size=M``:
+
+.. code-block:: html+django
+
+    {% query_string color=None %}
+
+If the given variable is a list, the value will remain as a list. If
+``my_list`` is set to ``["red", "blue"]``, the following code will result in
+``?color=red&color=blue``:
+
+.. code-block:: html+django
+
+    {% query_string color=my_list %}
+
+A common example of this tag is to preserve the current query string when
+displaying a page of results, while adding a link to the next and previous
+pages of results. For example, if the paginator is currently on page 3, and
+the current query string is ``?color=blue&size=M&page=3``, the following code
+would output ``?color=blue&size=M&page=4``:
+
+.. code-block:: html+django
+
+    {% query_string page=page.next_page_number %}
+
+You can also store the value in a variable, for example if you need multiple
+links to the same page with syntax such as
+``{% query_string page=page.next_page_number as next_page %}``.
```

Comment:
With the long inline code block this will probably wrap badly?

```suggestion
links to the same page with syntax such as:

.. code-block:: html+django

    {% query_string page=page.next_page_number as next_page_link %}
```